TECHNICAL-DECISIONS.md

```diff
@@ -0,0 +1,53 @@
+# Technical Decisions
+
+## This document
+
+This document lists decisions already made in this project so far, to make sure there is an explicit record of the work 
+we have done and for the benefit of new participants.
+
+## Git
+
+### New Work
+
+See CONTRIBUTING.md for more guidelines
+
+### Pull Requests
+
+Each Pull Request needs to be approved by one other person.
+
+Merging should be done using the "Create a merge commit" strategy.
+
+## Python
+
+### Python 3
+
+Python 3 is a minimum requirement.
+
+### PEP8
+
+Code formatting follows PEP8 with the exception of line lengths.
+
+flake8 is run in Travis. All code should be compliant before being sent for code review.
+
+## Database 
+
+### Postgres 10
+
+Postgres 10 is a minimum requirement, to take advantage of the latest features.
+
+### Table Names
+
+Table names are Singular.
+
+### ID columns
+
+ID columns are called "id", not including the name of the table. eg "release.id", not "release.release_id"
+
+## Automatic tests
+
+Tests are written using pytest and are run in Travis. All tests should pass before a code is sent for code review.
```

Comment:
Is the "all tests must pass before review" a decision that was agreed? I don't recall the discussion (but I wasn't around for every discussion!) and I've seen people open PRs to invite early comment, which I've assumed has been part of a 'working in the open' philosophy.

Reply:
There is a difference between "opening a PR for early comment" and "opening a PR for code review for merging". We discussed we would set GitHub to not allow merging till Travis passed, so that means the tests must pass before the later happens, but it's a fair point the tests don't need to pass before the former happens. 

Basically, I think we are getting confused on what "code review" here means.

To solve this, maybe we could add something like "It's fine to open a Work In Progress (WIP) pull request to invite early comments from your colleagues but make sure it's clearly marked as such."

Reply:
> To solve this, maybe we could add something like "It's fine to open a Work In Progress (WIP) pull request to invite early comments from your colleagues but make sure it's clearly marked as such."

That would make it very clear, thanks!

Reply:
Done! I'm starting to feel some content is awkwardly split between TECHNICAL-DECISIONS.md and CONTRIBUTING.md but maybe lets get this basic content in and then let others add to it and weight in? (That being the point of the doc after all!)